File: backend/App.DAL.DTO/Event.cs
using System.ComponentModel.DataAnnotations;
using App.Domain.Identity;
using Base.Domain;

namespace App.DAL.DTO;

public class Event : BaseEntityId
{
    public Guid AppUserId { get; set; }
    public AppUser? AppUser { get; set; }
    public DateTime StartTime { get; set; }
    
    [MaxLength(256)]
    public string Location { get; set; } = default!;
    
    [MaxLength(256)]
    public string Name { get; set; } = default!;
    
    [MaxLength(1000)]
    public string AdditionalInformation { get; set; } = default!;
    
    public ICollection<ParticipantEvent>? ParticipantEvents { get; set; }
}

File: frontend/entities/Event.ts
import ParticipantEvent from "./ParticipantEvent";

export default class Event {
  id: string;
  startTime: string;
  location: string;
  name: string;
  additionalInformation: string;
  participantEvents: ParticipantEvent[];

  constructor(
    id?: string,
    startTime?: string,
    location?: string,
    name?: string,
    additionalInformation?: string,
    participantEvents?: ParticipantEvent[]
  ) {
    if (
      id &&
      startTime &&
      location &&
      name &&
      additionalInformation &&
      participantEvents
    ) {
      this.id = id;
      this.startTime = startTime;
      this.location = location;
      this.name = name;
      this.additionalInformation = additionalInformation;
      this.participantEvents = participantEvents;
    } else {
      this.id = "";
      this.startTime = "";
      this.location = "";
      this.name = "";
      this.additionalInformation = "";
      this.participantEvents = [];
    }
  }
}
